Modules/Core/Common/include/itkMetaProgrammingLibrary.h

```diff
@@ -210,6 +210,19 @@ struct Not : NotC<TF::Value>
   using Type = typename NotC<TF::Value>::Type;
 };
 
+/** MPL type trait to check if type is a SmartPointer.
+ */
+template <typename T>
+struct IsSmartPointer : FalseType
+{};
+
+
+/// \cond SPECIALIZATION_IMPLEMENTATION
```

Comment:
What is this for?

Reply:
Only the non-specialized class should show up in doxygen, so this removes the specialization for Doxygen.